services/datagrid-service.json

```diff
@@ -0,0 +1,442 @@
+                                    "key" : "application-password"
+                                 }
+                              }
+                           },
+                           {
+                              "name": "DEFAULT_OWNERS",
+                              "value": "1"
+                           },
+                           {
+                              "name": "DEFAULT_CACHE_PARTITION_HANDLING_WHEN_SPLIT",
+                              "value": "DENY_READ_WRITES"
+                           },
+                           {
+                              "name": "DEFAULT_CACHE_PARTITION_HANDLING_MERGE_POLICY",
+                              "value": "REMOVE_ALL"
+                           }
+                        ],
+                        "image": "${IMAGE}",
+                        "livenessProbe": {
+                           "exec": {
+                              "command": [
+                                 "/opt/datagrid/bin/livenessProbe.sh"
+                              ]
+                           },
+                           "initialDelaySeconds": 60,
+                           "timeoutSeconds": 10,
+                           "periodSeconds": 20,
+                           "successThreshold": 1,
+                           "failureThreshold": 5
+                        },
+                        "readinessProbe": {
+                           "exec": {
+                              "command": [
+                                 "/opt/datagrid/bin/readinessProbe.sh"
+                              ]
+                           },
+                           "initialDelaySeconds": 60,
+                           "timeoutSeconds": 10,
+                           "periodSeconds": 20,
+                           "successThreshold": 2,
+                           "failureThreshold": 5
+                        },
+                        "name": "${APPLICATION_NAME}",
+                        "ports": [
+                           {
+                              "containerPort": 8080,
+                              "name": "http",
+                              "protocol": "TCP"
+                           },
+                           {
+                              "containerPort": 8888,
+                              "name": "ping",
+                              "protocol": "TCP"
+                           },
+                           {
+                              "containerPort": 11222,
+                              "name": "hotrod",
+                              "protocol": "TCP"
+                           }
+                        ],
+                        "volumeMounts": [
+                           {
+                              "name": "srv-data",
+                              "mountPath": "/opt/datagrid/standalone/data"
+                           },
+                           {
+                              "name": "srv-data",
```

Comment:
The volumemounts and volume claims for the keystores need to be updated. I ran into problems with these and the scaling of stateful sets. See the merged `caching-service.json` for specifics.